### ISSUE TITLE
Text editor - option to manually or automatically refresh the view

### ISSUE THREAD

**Issue by oliveru** [HeidiSQL/HeidiSQL#92]
## Expected behavior

Wish when a large column opens in a text editor, I could easily choose between some refresh rates because I often watch logs being written to the database. Also automatically scroll to the end with this option so I see which data is written into that column in "real time". Also manually refreshing inside the text editor view would be nice.

## Current behavior

Currently I have to close the text editor, refresh the query and open the column again and scroll down to see new data.

### Steps to reproduce

1. Open a large column with lots of data being written into it, as in a scheduler event
2. The text editor will open
3. You can't refresh manually or automatically

### Context

Please provide any relevant information about your setup. This is important in case the issue is not reproducible except for under certain conditions.

* HeidiSQL version: 9.5.0.5196
* Database system + version: MySQL 5.7
* Operating system: Windows 10 64bit

### Failure Logs

**Comment by fifonik** [comment]
How you suggests to merge data that is in editor with the data that comes from DB during refresh?

I do not want that text editor's data is updated automatically with even a little chance to lost something that I've just keyed in in the text editor field. So if the feature to be implemented, I thing it should be an option to switch it on and off and it must be switched off by default.

P.S. I do not know your scenario and I could be wrong, however, I'd say you using your SQL server in a very strange way. Keeping logs as a text in one table field is something that you should not do with SQL. You should add different records instead: one log record == one record in SQL table with at least two columns: time stamp and text.

**Comment by oliveru** [comment]
There is a schuduler event table. The events can write logs in their row with execution start/end. Common pratice. So if it is a long running event, logs are being inserted in from that running task like.

[id] success
[id] error: ...
[id] ...

Just to give you an idea.
So if I watch the log, I have to refresh the text editor data. Would be nice to stay in the text editor. That's all.

**Comment by rentalhost** [comment]
Hm... I don't know. You mean an auto-refresh? And if the user don't like to really refresh it anytime? Maybe a F5 to force refresh be more useful?

**Comment by oliveru** [comment]
Yes it should be in the users hand. I would definetly be happy if I could just manually refresh the view without closing it. An Auto - Refresh would be a nice addition.

**Comment by rentalhost** [comment]
Oh, I think that it could be done, but not automatically. Maybe time in time the Text Editor checks if the cell was modified (ex. 5s/5s), if do, display some info on status bar like "Content was modified remotely, click to update". And another option to refresh manually (like an F5).

What do you think?

**Comment by oliveru** [comment]
Sounds very nice, in my case I don't need the info that the data has changed but refreshing the text editor manually would be very handy.

**Comment by Ejsstiil** [comment]
Wanted to ask about this feature? I'd like be happy about this feature, and if I may add one, it would be nice if the just-opened editor/viewer does not pre-select all its content. Maybe configurable for those who got used to this behaviour.
Thank you!

**Comment by Ejsstiil** [comment]
> Oh, I think that it could be done, but not automatically. Maybe time in time the Text Editor checks if the cell was modified (ex. 5s/5s), if do, display some info on status bar like "Content was modified remotely, click to update". And another option to refresh manually (like an F5).
> 
> What do you think?

F5 wants to do a refresh: when no changes made - refresh,
otherwise ask to discard/save current changes (with full cell overwrite), then refresh
